### ISSUE TITLE
Key(s) sent on Steam button

### ISSUE THREAD

**Issue by bartervg** [bartervg/barter.vg#256]
# What problem does this feature address?
After accepting an offer and Steam request, it can be convenient to asynchronously exchange keys. To increase the likelihood that the other side notices this, many senders respond in the offer `key sent on Steam`. However, for those that sent the offer, despite fulfilling the offer, and now waiting for the other side to respond, the offer remains is a need to act status for the offer sender.

## Describe a solution
A button near chat `Key(s) sent on Steam`. Clicking this would change the side to act and thus remove the need to act on offer icon. The key sender has nothing left to do but wait for the other side to respond.
![image](https://user-images.githubusercontent.com/28576132/113440467-7c16c800-93ba-11eb-8682-62292a8c71e8.png)

![image](https://user-images.githubusercontent.com/28576132/113440195-eda24680-93b9-11eb-8fb5-e37a6d8367a2.png)

There could be a preference for those that do not want keys sent until chat confirmation. This would disable such button.

![image](https://user-images.githubusercontent.com/28576132/113440331-3823c300-93ba-11eb-9694-db3cc93afb2a.png)

In addition, there could be a button for Steam invite sent.

## Examples of similar features
Preference on when to send Steam invite and minimum offer expiration days.

**Comment by Luckz** [comment]
eBay-like, an "I completed my part of the trade" (= payment sent / goods shipped)!
Of course it has to be optional to use this and not look like it's part of the proper etiquette, since it's only useful for async trades.

Normally it's the accepting partner who sends a friend request the moment they accept the trade, innit? Perhaps there's cases where people are at work or in the mountains and have no access to their Steam account, then separating the invite makes sense I guess. It doesn't seem common enough to show a button about it for everyone. Not sure if a __preference__ to add such an exotic button is worth the additional cognitive load on our *code output creatures*.

If a user set a preference to refuse _key ambushes_, the other partner should see the disabled button say as much.

`I sent key(s) on Steam`
`confirm trade in chat before sending keys!`

**Comment by Revadike** [comment]
Yes, please make it optional. I'm lazy and don't want to change my trade bot to add this extra step during trading.
Personally, I don't get it. It's pretty much the same as typing the message "I sent steam keys" but then in button form.

**Comment by bartervg** [comment]
> 
> 
> Yes, please make it optional. I'm lazy and don't want to change my trade bot to add this extra step during trading.

Yes, it would be optional.

> Personally, I don't get it. It's pretty much the same as typing the message "I sent steam keys" but then in button form.

The difference is that it would change the side to act. For example, if I've sent Steam invites and keys and I'm waiting for everyone to respond. I shouldn't see ![image](https://user-images.githubusercontent.com/28576132/113799162-8e7a6400-9722-11eb-94ce-fd0708f6d279.png). It should be the users that needs to act that see that.

**Comment by Revadike** [comment]
It's worth mentioning that `Key(s) sent on Steam` is not a very good name for this button. It's too specific and doesn't cover all cases. Something like `Expected items sent to other party` applies to all cases.
